src/Functions.php

```diff
@@ -60,7 +61,17 @@ function beforeEach(Closure $closure = null): BeforeEachCall
      */
     function dataset(string $name, Closure|iterable $dataset): void
     {
-        DatasetsRepository::set($name, $dataset);
+        $file = Backtrace::datasetsFile();
+        $filename = Str::afterLast($file, DIRECTORY_SEPARATOR);
+        $scope = Str::beforeLast($file, DIRECTORY_SEPARATOR);
+
+        if (Str::afterLast($scope, DIRECTORY_SEPARATOR) === 'Datasets') {
```

Comment:
We do these same `Str` checks in multiple files. I wonder if we should introduce a `Support\Dataset` class?

Reply:
@lukeraymonddowning I refactored this a bit, extracted to a support class and changed Str:: support functions with dirname and basename, so there is less code to maintain

you can find the changes here: 

https://github.com/pestphp/pest/pull/584/commits/80854d5f8700a958be28433cd60c2de74404c4a3

Reply:
So much nicer ❤️